### ISSUE TITLE
Wrong `Authorization` header is used when refreshing tokens

### ISSUE THREAD

**Issue by osaxma** [supabase-community/gotrue-dart#52]
The issue is better described here: 
- https://github.com/supabase/supabase/issues/5110 

TL;DR: recovering session returns "Invalid Refresh Token" error. 

## Root cause

I debugged the issue manually using the REST API for go-true and found no issues with the server. I also debugged gotrue-dart and I believe I found the root cause there.

When making a refresh token request to gotrue server, the `Authorization` header should be the `JWT` from the recovered session. Since a `SupabaseClient` is instantiated before recovering the session, the `Authorization` header defaults to the `apikey`. Therefore, when calling `client.auth.receoverSession(sessionString)`, the `Authorization` header will be incorrect and it doesn't get updated with the JWT which causes an "Invalid Refresh Token" and I believe it invalidates the token from being used in the future as well. 

## To reproduce:
- place a debug point here:
https://github.com/supabase-community/gotrue-dart/blob/936e807159658539ee95d83925e297c900b05fa2/lib/src/gotrue_api.dart#L357

- Invoke `client.auth.receoverSession(sessionString)` with an expired JWT and valid refresh token.
- Print the headers, you'll find the Authorization isn't the JWT from the recovered session:
```
{
  "X-Client-Info" : "supabase-dart/0.2.14",
  "apikey" : "<api_key>",
  "Authorization" : "Bearer <api_key>" <~~~ this should be the JWT
}
```
- Also the result of `client.auth.receoverSession(sessionString)` should have an "Invalid Refresh Token" error.

## Additional Context: 
I believe the same issue happens when a token is refreshed before it's expired by the timer because I always have to log-in again after 1-hour.

**Comment by osaxma** [comment]
FYI: the following bug on gotrue server could also be another reason for this issue: 

- https://github.com/supabase/gotrue/issues/348
